dts/bindings/base/zephyr,memory-region.yaml

```diff
@@ -16,3 +16,11 @@ properties:
       memory region in the final executable. The region address and size
       is taken from the <reg> property, while the name is the value of
       this property.
+
+  zephyr,memory-region-mpu:
```

Comment:
I think it makes more sense for this to be a string `enum`. It turns a runtime error into a devicetree parse error. Like this if unaware:
https://github.com/zephyrproject-rtos/zephyr/blob/0521aa0fbce835022fbd65d0d438f4a20d5af893/dts/bindings/power/zephyr%2Cpower-state.yaml#L9-L17

Unless it is expected that individual MPU drivers are going to be defining a whole mess of incompatible options, which seems unlikely.

Reply:
>  which seems unlikely.

If true, then I agree here

Reply:
yeah, I'll try to move to enums. I'm still not 100% sure this is feasible due to some restrictions on how the struct field are built but I'll keep you posted ;)

Reply:
AFAIK it's no different from what you're doing now apart from restricting the values the user can input in devicetree.
I'm not suggesting moving away from the "property->callable macro" setup you have now.

Reply:
Ooooh, sorry, I misunderstood your point. Yes, this is a good idea indeed.

Reply:
done